py3status/modules/mysql_slave.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+"""
+Display MySQL/MariaDB slave state.
+
+This module displays the number of seconds the slave is behind the master.
+
+Configuration parameters:
+    cache_timeout: refresh cache_timeout for this module (default 5)
+    format: display format for this module
+        *(default '\?color=seconds_behind_master {host} is '
+        '[\?if=seconds_behind_master {seconds_behind_master}s behind|master]') *
+    host: the host to connect to (default None)
+    passwd: the password to use for the connection (default None)
+    port: the port to connect to (default 3306)
+    query: The sql query to run (default 'show slave status')
+    thresholds: a threshold list of tuples, -1 used for a master hosts
+        (default {'seconds_behind_master':
+            [(0, 'deepskyblue'), (100, 'good'), (300, 'degraded'), (600, 'bad')]
+        })
+    user: the user used for the connection (default None)
+
+Requires:
+    MySQLdb: python package for python
+
+@author cereal2nd
+@license BSD
+
+SAMPLE OUTPUT
+{'full_text': 'test is 0s behind'}
+{'full_text': 'test is 509s behind'}
+{'full_text': 'test is master'}
+"""
+
+from __future__ import absolute_import
+from MySQLdb import connect
```

Comment:
If you add this:
```
from importlib import import_module
```
and...
```
    def post_config_hook(self):
        if not self.name:
            raise Exception('missing name')
        if not self.parameters:
            raise Exception('missing parameters')
        self.connect = getattr(import_module(self.name), 'connect')
```
then we should be able to use any sql db... so maybe rename this module to `sql` now too. :-) 

P.S. do we need `from __future__ import absolute_import` too?

Reply:
Hmm, i have no way to test this out, so not sure if we can simply do this.
Do all other db engines use the same connect method?

Reply:
Just tested this module. Fine with `name = 'sqlite3'` --> `from sqlite3 import connect` (from `thunderbird_todos`). The answer *seems* like yes to me.

Reply:
would be type a better name instead of 'name'?

Reply:
I think `name` because we can `format = '{name}'` --> `sqlite3`. Also, the name `name` is coming from `importlib` documentation / parameter names.

Reply:
Maybe `database` would be even better... `database` -> `name` -> `type` (imho).
```
sql {
    database = 'MySQLdb'
    format = '{placeholder1} {placeholder2} {placeholder3}'
    query = 'SELECT * FROM ...'
    parameters = {
        'host': 'empty',
        'passwd': 'empty',
    }
}